tests/ludwig/config_validation/test_checks.py

```diff
@@ -117,3 +118,21 @@ def test_comparator_fc_layer_config(
 
     with pytest.raises(ConfigValidationError) if not expect_success else contextlib.nullcontext():
         ModelConfig.from_dict(config)
+
+
+def test_dense_binary_encoder_1_layer():
```

Comment:
This says it's a test for 1-layer encoders, but `num_layers` is set to 0. Is that correct? If so, could you add a docstring explaining the test and why we expect this config to be invalid?